### PR TITLE
TEN-154 added in auth connector for deprecated route

### PR DESCRIPTION
**Pull request by ChrisHillCap**

### CODE REVIEW

**Comment by ChrisHillCap** [comment]
https://github.com/hmrc/auth/blob/51d78de48c0a9aa11095058bce0d78d00f20c3d1/it/auth/legacy/authority/LegacyAuthorityControllerISpec.scala#L358
https://github.com/hmrc/auth/blob/09e3b831752bfb0e820be4361806ba7b7f3051a5/app/auth/legacy/authority/AuthorityController.scala#L80

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/config/AppConfig.scala [hunk 1]
```diff
@@ -33,6 +33,8 @@ class AppConfig @Inject()(val config: ServicesConfig) {
     : String = config.baseUrl("enrolment-store-proxy") + "/enrolment-store-proxy"
   lazy val TAX_ENROLMENTS_BASE_URL
     : String = config.baseUrl("tax-enrolments") + "/tax-enrolments"
+  lazy val AUTH_BASE_URL
+  : String = config.baseUrl("auth")
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/config/AppConfig.scala:37]: Is this needed? we already get this as a part of the AuthConnector anyway.

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/connectors/LegacyAuthConnector.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  def updateEnrolments(enrolments: Set[Enrolment])(
+                        implicit ec: ExecutionContext,
+                        hc: HeaderCarrier): TEAFResult[Unit] = EitherT {
+    httpClient
+      .PUT[Set[Enrolment], HttpResponse](s"${appConfig.AUTH_BASE_URL}/auth/enrolments", enrolments)(wts = EnrolmentsFormats.writes, implicitly, implicitly, implicitly)
+      .map(httpResponse =>
+        httpResponse.status match {
+        case OK => Right(Unit)
+        case _ =>
+          logger.logEvent(LoggingEvent.logUnexpectedErrorFromAuthWhenUsingLegacyEndpoint(httpResponse.status))
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/connectors/LegacyAuthConnector.scala:50]: Can the response body be added to the error log? Could be helpful for diagnosing.

Comment (ChrisHillCap) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/connectors/LegacyAuthConnector.scala:50]: theres no response body, its always empty

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/config/AppConfig.scala (hunk 1 above)
Comment (ChrisHillCap) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/config/AppConfig.scala:37]: i thought this, but then, youd have to inject the AuthConnector which seems like a weird dependency 🤷

**Review by Ufuoma-Daniel — APPROVED** [review]

**Pull request merged**